src/backend/graphics/opengl.rs

```diff
@@ -0,0 +1,17 @@
+use nix::c_void;
+
+pub enum SwapBuffersError {
+    ContextLost,
+    AlreadySwapped,
+}
+
+pub trait OpenGLRenderer
+{
+    fn swap_buffers(&self) -> Result<(), SwapBuffersError>;
+    unsafe fn get_proc_address(&self, symbol: &str) -> *const c_void;
+    fn get_framebuffer_dimensions(&self) -> (u32, u32);
+    fn is_current(&self) -> bool;
+    unsafe fn make_current(&self);
+    fn get_api(&self) -> Api;
+    fn get_pixel_format(&self) -> PixelFormat;
```

Comment:
I am also in favour of implementing traits of said opengl libraries. This will ease integration but most importantly always make sure at least the advertised libraries are working on the trait. If the situation gets to complicated at any time, we can still discuss dropping it.